File: userservice/src/main/java/hu/bme/aut/classifiedadvertisementsite/userservice/service/messagequeue/NotificationMessageQueue.java
package hu.bme.aut.classifiedadvertisementsite.userservice.service.messagequeue;

import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

@Component
public class NotificationMessageQueue {
    private static final String QUEUE_NAME = "email-queue";

    @Bean
    Queue queue() {
        return new Queue(QUEUE_NAME, true);
    }
}

File: bidservice/src/main/kotlin/hu/bme/aut/classifiedadvertisementsite/bidservice/service/messagequeue/NotificationMessageQueue.kt
package hu.bme.aut.classifiedadvertisementsite.bidservice.service.messagequeue

import org.springframework.amqp.core.Queue
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component

@Component
class NotificationMessageQueue {
    companion object {
        const val EMAIL_QUEUE_NAME = "email-queue"
        const val PUSH_QUEUE_NAME = "push-queue"
    }

    @Bean("email-queue")
    fun emailQueue(): Queue {
        return Queue(EMAIL_QUEUE_NAME, true)
    }

    @Bean("push-queue")
    fun pushQueue(): Queue {
        return Queue(PUSH_QUEUE_NAME, true)
    }
}